server/src/domain/domain.config.ts

```diff
@@ -14,16 +14,23 @@ export const immichAppConfig: ConfigModuleOptions = {
   isGlobal: true,
   validationSchema: Joi.object({
     NODE_ENV: Joi.string().optional().valid('development', 'production', 'staging').default('development'),
+    LOG_LEVEL: Joi.string()
+      .optional()
+      .valid(...Object.values(LogLevel)),
+
     DB_USERNAME: WHEN_DB_URL_SET,
     DB_PASSWORD: WHEN_DB_URL_SET,
     DB_DATABASE_NAME: WHEN_DB_URL_SET,
     DB_URL: Joi.string().optional(),
     DB_VECTOR_EXTENSION: Joi.string().optional().valid('pgvector', 'pgvecto.rs').default('pgvecto.rs'),
-    LOG_LEVEL: Joi.string()
-      .optional()
-      .valid(...Object.values(LogLevel)),
+
     MACHINE_LEARNING_PORT: Joi.number().optional(),
     MICROSERVICES_PORT: Joi.number().optional(),
-    SERVER_PORT: Joi.number().optional(),
+    IMMICH_METRICS_PORT: Joi.number().optional(),
+
+    IMMICH_METRICS: Joi.boolean().optional().default(false),
+    IMMICH_HOST_METRICS: Joi.boolean().optional().default(false),
+    IMMICH_API_METRICS: Joi.boolean().optional().default(false),
+    IMMICH_IO_METRICS: Joi.boolean().optional().default(false),
```

Comment:
Another option could be like `IMMICH_METRICS=host,io,api`.

Reply:
I tried this but didn't really like it. Joi will only accept JSON arrays, so the input either has to be `["host", "io", "api"]` or the validation has to be a gnarly regex like `/^((((api)|(host)|(io))(($)|(, ?))))+$/`. I think the separate envs are cleaner.